crates/database/src/orders.rs

```diff
@@ -206,6 +213,28 @@ WHERE uid = $1
     sqlx::query_as(QUERY).bind(id).fetch_optional(ex).await
 }
 
+/// Given a list of order ids, return the orders that are present in the
+/// database.
+pub async fn orders_that_exist(
+    ex: &mut PgConnection,
+    order_ids: &[OrderUid],
```

Comment:
How about using an Iterator to avoid additional collect operations?
```suggestion
    order_ids: impl Iterator<Item = OrderUid>,
```